packages/cloudflare/data_stream/logpull/elasticsearch/ingest_pipeline/http.yml

```
@@ -0,0 +1,441 @@
+---
+description: Pipeline for parsing cloudflare http logs
+processors:
+# Event Time Fields
+- convert:
+    field: json.EdgeStartTimestamp
+    type: string
+- convert:
+    field: json.EdgeEndTimestamp
+    type: string
+- gsub:
+    field: json.EdgeStartTimestamp
+    pattern: "\\d{6}$"
+    replacement: ""
+    if: "ctx?.json?.EdgeStartTimestamp != null && (ctx?.json?.EdgeStartTimestamp).length() > 18"
+- gsub:
+    field: json.EdgeEndTimestamp
+    pattern: "\\d{6}$"
+    replacement: ""
+    if: "ctx?.json?.EdgeEndTimestamp != null && (ctx?.json?.EdgeEndTimestamp).length() > 18"
+- date:
+    field: json.EdgeStartTimestamp
+    formats:
+    - ISO8601
+    - uuuu-MM-dd'T'HH:mm:ssX
+    - uuuu-MM-dd'T'HH:mm:ss.SSSX
+    - yyyy-MM-dd'T'HH:mm:ssZ
+    - yyyy-MM-dd'T'HH:mm:ss.SSSZ
+    - UNIX_MS
+    timezone: UTC
+    target_field: "@timestamp"
+- date:
+    field: json.EdgeStartTimestamp
+    formats:
+    - uuuu-MM-dd'T'HH:mm:ssX
+    - uuuu-MM-dd'T'HH:mm:ss.SSSX
+    - yyyy-MM-dd'T'HH:mm:ssZ
+    - yyyy-MM-dd'T'HH:mm:ss.SSSZ
+    - UNIX_MS
+    timezone: UTC
+    target_field: "event.start"
+- date:
+    field: json.EdgeEndTimestamp
+    formats:
+    - uuuu-MM-dd'T'HH:mm:ssX
+    - uuuu-MM-dd'T'HH:mm:ss.SSSX
+    - yyyy-MM-dd'T'HH:mm:ssZ
+    - yyyy-MM-dd'T'HH:mm:ss.SSSZ
+    - UNIX_MS
+    timezone: UTC
+    target_field: "event.end"
+- script:
+    lang: painless
+    if: ctx?.event?.start != null && ctx?.event?.end != null
+    source: >-
+      ZonedDateTime start = ZonedDateTime.parse(ctx.event.start);
+      ZonedDateTime end = ZonedDateTime.parse(ctx.event.end);
+      ctx.event.duration = ChronoUnit.NANOS.between(end, start);
+# Source Fields
+- rename:
+    field: json.ClientIP
+    target_field: source.address
+    ignore_missing: true
+- convert:
+    field: source.address
+    target_field: source.ip
+    type: ip
+    ignore_missing: true
+    ignore_failure: true
+- geoip:
+    field: source.ip
+    target_field: source.geo
+    ignore_missing: true
+    if: "ctx.source?.ip != null"
```

```suggestion
    ignore_missing: true
```

I think this is redundant.